src/redux/features/observations/observationPlantingZoneSelectors.ts

```diff
@@ -0,0 +1,29 @@
+import { createCachedSelector } from 're-reselect';
+import { createSelector } from '@reduxjs/toolkit';
+import { RootState } from 'src/redux/rootReducer';
+import { ObservationPlantingZoneResults } from 'src/types/Observations';
+import { SearchParams, DetailsParams, selectObservationDetails } from './observationDetailsSelectors';
+import { searchResultPlots } from './utils';
+
+export type ZoneParams = DetailsParams & {
+  plantingZoneId: number;
+};
+
+export type ZoneSearchParams = SearchParams & ZoneParams;
+
+export const selectObservationPlantingZone = createSelector(
+  [selectObservationDetails, (state, params, defaultTimeZone) => params],
+  (observationDetails, params) =>
+    observationDetails?.plantingZones.find(
+      (plantingZone: ObservationPlantingZoneResults) => plantingZone.plantingZoneId === params.plantingZoneId
+    )
+);
+
+export const searchObservationPlantingZone = createCachedSelector(
+  selectObservationPlantingZone,
+  (state: RootState, params: ZoneSearchParams, defaultTimeZone: string) => params,
+  (plantingZone, params) => searchResultPlots(params.search, plantingZone)
+)(
+  (state: RootState, params: ZoneSearchParams, defaultTimeZone: string) =>
+    `${params.plantingSiteId}_${params.observationId}_${params.plantingZoneId}_${defaultTimeZone}_${params.search}`
```

Comment:
If I understand correctly, this string is constructed of the possible fields we want to search against, and then we regexp match against it?

Reply:
Oh, this is the key for the cache. The cached selector returns cached responses so it doesn't have to recalculate.
It recalculates only if the key is missing (new input params such as planting site id, etc.) or if any of the intermediate data changes.
```
  createCachedSelector(
    (state, args) => intermediate selector 1,  // intermediate selectors
     ...
    (state, args) => intermediate selector n,
    (results-1, ... results-n) => return final response, // arguments to this last function are results from above intermediate selectors
  ) (state, args) => { return cache key for this response }
```
This is provided by `re-reselect` an extension of redux's `cachedSelector`.
If we don't used cached selectors, the results generated will have different reference values each time and the component will render a lot.

Reply:
Here's a good [article](https://redux.js.org/usage/deriving-data-selectors#writing-memoized-selectors-with-reselect) explaining memoized selectors with mention of  [re-select](https://redux.js.org/usage/deriving-data-selectors#re-reselect) as well.

Reply:
@charlie-tf if it helps, there are more cached selector use-cases coming up, happy to assign one to you. Let me know.

Reply:
Definitely would love to work on one. Thanks for the article

Reply:
Charlie.. assigned https://terraformation.atlassian.net/browse/SW-3748 and https://terraformation.atlassian.net/browse/SW-3746 to you. Whichever you'd like to do, feel free to reassign/un-assign as per your plans/bandwidth.